src/service/performance-impl.js

```diff
@@ -213,6 +214,38 @@ export class Performance {
     observer.observe({entryTypes: ['paint']});
   }
 
+  /**
+   * Reports first input delay.
+   * See https://wicg.github.io/event-timing/
+   */
+  registerEventTimingObserver_() {
+    if (!this.win.PerformanceEventTiming) {
+      return;
+    }
+    // Chrome doesn't implement the buffered flag for PerformanceObserver.
+    // That means we need to read existing entries and maintain state
+    // as to whether we have reported a value yet, since in the future it may
+    // be reported twice.
+    // https://bugs.chromium.org/p/chromium/issues/detail?id=725567
+    let recordedFirstInputDelay = false;
+    const processEntry = entry => {
+      if (recordedFirstInputDelay) {
+        return;
+      }
+      this.tick('fid', entry.processingStart - entry.startTime);
```

Comment:
This line is throwing this error in Travis CI:

```
src/service/performance-impl.js:235: ERROR - Property processingStart never defined on entry
      this.tick('fid', entry.processingStart - entry.startTime);
                             ^^^^^^^^^^^^^^^
```

I tried adding an annotation of `PerformanceEntry` for the `entry` parameter like this:

```js
/**
 * @param {PerformanceEntry} entry
 */
```

However, the `processingStart` property is part of the [new `PerformanceEventTiming` interface](https://wicg.github.io/event-timing/#sec-performance-event-timing), and doesn't seem to be defined on the `PerformanceEntry` type. Which makes sense - the Event Timing API is in Editor's Draft status.

I tried specifying an unknown type inline:

```js
this.tick('fid', (/** @type {?} */ (entry)).processingStart - entry.startTime);
```

but the compiler didn't like this either:

```
src/service/performance-impl.js:235: ERROR - Property processingStart never defined on ?
      this.tick('fid', (/** @type {?} */ (entry)).processingStart - entry.startTime);
                                                  ^^^^^^^^^^^^^^^
```

Any suggestion here?

Reply:
@erwinmombay should this be added to closure compiler extern (I actually don't know where it is) or amp.extern.js?

Reply:
I created a separate extern for these definitions for now. 

[Maybe](https://github.com/google/closure-compiler/issues/3233) this should be added to closure-compiler itself - I'll follow-up there if so. Defining our own extern in this project allows us to move forward with this feature for the moment.